unit2-csharp/lesson12.rst

```diff
@@ -10,12 +10,11 @@ Goals
 Objectives
 ----------
 
-- Use C# Validation API annotations (from ``C#x.validation.constraints``) on model classes, including ``@Min``, ``@Max``, ``@Size``, ``@Email``, ``@NotBlank``, and ``@NotNull``
-- Use ``@Valid`` on model objects in a controller to check for validation errors
-- Explain how and when Spring validates model objects that are bound during a request
-- Display error messages in the view
-- Use ``th:field`` to auto-generate field settings
-- Use ``th:errors`` to display field-specific error messages 
+- Describe how ViewModels work and why you would want to use them
+- Use ``asp-for`` and ``asp-validation-for`` in your view to utilize the ViewModel
+- Explain how you can use ViewModels to add a layer of validation
+- Use the ``[Required]`` attribute
```

Comment:
I'm wondering if there are other [built-in validation](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1#built-in-attributes) attributes that would be good to expose learners to. I'm not sure which ones would fit well within the coding events app, but `[Range]`, `[StringLength]`, and `[EmailAddress]` seem like good, commonly-used candidates.

Reply:
One thought is to see what we did for the corresponding Java lesson.